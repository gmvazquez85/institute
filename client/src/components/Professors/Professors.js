import React, { Component } from "react";

class Professors extends Component {
  
  render() {
    let { professors } = this.props;

    professors = professors ? professors : [];

    return (
      <div>
        
        <ul>
          {professors.length <= 0
            ? "NO DB ENTRIES YET"
            : professors.map(dat => (
                <li style={{ padding: "10px" }} key={dat.id}>
                  <span style={{ color: "gray" }}> id: </span> {dat.id} <br />
                  <span style={{ color: "gray" }}> Name: </span> {dat.name} <br />
                  {/*< button onClick={() => this.edit(dat.id)}>EDIT</button> <br />*/}
                  {/*< button onClick={() => this.deleteFromDB(dat.id)}>DELETE</button> <br />*/}
                </li>
              )
            )
          }
        </ul>
      </div>
    );
  }
}

export default Professors;