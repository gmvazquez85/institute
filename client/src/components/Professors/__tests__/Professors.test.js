import React from 'react';
import { shallow } from 'enzyme';
import Professors from '../Professors';

describe('Professors', () => {
  it('should render correctly without professors', () => {
    const component = shallow(<Professors />);
  
    expect(component).toMatchSnapshot();
  });
  
  it('should render correctly with professors', () => {

  	const professors = [{id:1, name: "gabriel"},{id:2, name:"matias"}];

    const component = shallow(<Professors professors={professors}/>);
  
    expect(component).toMatchSnapshot();
  });
});