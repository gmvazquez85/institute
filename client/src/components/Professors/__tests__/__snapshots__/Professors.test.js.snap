// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Professors should render correctly with professors 1`] = `
<div>
  <ul>
    <li
      key="1"
      style={
        Object {
          "padding": "10px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         id: 
      </span>
       
      1
       
      <br />
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         Name: 
      </span>
       
      gabriel
       
      <br />
    </li>
    <li
      key="2"
      style={
        Object {
          "padding": "10px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         id: 
      </span>
       
      2
       
      <br />
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
         Name: 
      </span>
       
      matias
       
      <br />
    </li>
  </ul>
</div>
`;

exports[`Professors should render correctly without professors 1`] = `
<div>
  <ul>
    NO DB ENTRIES YET
  </ul>
</div>
`;
